capability/capability_test.go

```diff
@@ -21,17 +21,31 @@ const (
 	maxLastCap = CAP_CHECKPOINT_RESTORE
 )
 
-func TestLastCap(t *testing.T) {
-	last, err := LastCap()
+// checkErrLinux does two things. First, it fails the test if err is:
+//   - not nil on Linux;
+//   - nil on other platforms.
+//
+// Returns true for Linux, false for other platforms.
+func checkErrLinux(t testing.TB, err error) bool {
```

Comment:
Hm.. so I'm a bit on the fence here; it feels like the helper abstraction only saves a few lines of code, but at the cost of now having to explain what it does (which is not immediately obvious by _only_ looking at the calls in the `TestXXX` functions), and now the `t.Fatal()` has to explain that it was non-linux because the test-failure would now show the line-number of the call-site, which is either Linux or non-Linux.

At least for me, it's _more_ cognitive load than the few lines saved.

We're also not checking for a specific error, which is something we could consider (looks like the total number of lines is even shorter than with the helper).

Finally, I noticed we're using `t.Fatal` in various places in this file; looking at most places, we could use `t.Error` instead, and not fail early; doing so _could_ still provide more context about the failing test, which could help estimate what the issue is.


Here's a quick draft what it _could_ look like without the helper;

```go
func TestLastCap(t *testing.T) {
	last, err := LastCap()
	if runtime.GOOS != "linux" {
		if !errors.Is(err, errNotSup) {
			t.Errorf("expected %v, got %v", errNotSup, err)
		}
		return
	}

	// Sanity checks (Linux only).
	if err != nil {
		t.Errorf("expected nil error, got %v", err)
	}
	if last < minLastCap {
		t.Errorf("LastCap returned %d (%s), expected >= %d (%s)", last, last, minLastCap, minLastCap)
	}
	if last > maxLastCap {
		t.Errorf("LastCap returned %d, expected <= %d (%s). Package needs to be updated.", last, maxLastCap, maxLastCap)
	}
}

func TestListSupported(t *testing.T) {
	supported, err := ListSupported()
	if runtime.GOOS != "linux" {
		if !errors.Is(err, errNotSup) {
			t.Errorf("expected %v, got %v", errNotSup, err)
		}
		return
	}

	// Sanity check (Linux only).
	if err != nil {
		t.Errorf("expected nil error, got %v", err)
	}
	t.Logf("got +%v (len %d)", supported, len(supported))
	minLen := int(minLastCap) + 1
	if len(supported) < minLen {
		t.Errorf("result is too short (got %d, want %d): +%v", len(supported), minLen, supported)
	}
}
```

Reply:
1. There's also a third test case which uses the same helper, and I expect more will be added. Also, we might have to add "android" next to "linux", and without a helper we'll have to do it in multiple places, but I'm with you.

2. Re: using `errNotSup` -- it's an internal ID, not exposed through the API, so I don't think it makes sense to check for it.

3. Re: this code:
```go
	// Sanity check (Linux only).
	if err != nil {
		t.Errorf("expected nil error, got %v", err)
	}
	t.Logf("got +%v (len %d)", supported, len(supported))
	minLen := int(minLastCap) + 1
	if len(supported) < minLen {
```

The first `t.Errorf` should be `t.Fatalf` because if List returned an error, we don't want to check the list length. Same in another test.

4. I used `want, got` instead of `expected, got` as I picked it up from Go sources, it's kind of a standard in there and they don't like other working. I find it short(er) and convenient.

5. `runtime.GOOS` in an error message is a subtle reminder why we expect non-nil error (because "Windows" etc).

6. I feel these changes are well beyond the scope of adding codespell, so they deserve a separate PR. Removed non-trivial changes from this PR, will open a new one soon.

Reply:
@thaJeztah ptal #162